frontend/test/pytest/test_aws_braket_devices.py

```diff
@@ -0,0 +1,644 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+pytest.importorskip("braket")
+
+import numpy as np
+import pennylane as qml
+
+from catalyst import grad, qjit
+
+
+class TestBraketGates:
+    @pytest.mark.parametrize(
+        "device",
+        [
+            qml.device(
+                "braket.aws.qubit",
+                device_arn="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
+                wires=3,
+            ),
+            qml.device(
+                "braket.local.qubit",
+                backend="default",
+                wires=3,
+            ),
+            qml.device(
+                "braket.local.qubit",
+                backend="braket_sv",
+                wires=3,
+            ),
+        ],
+    )
+    def test_no_parameters_braket(self, device):
+        """Test no-param operations on braket devices."""
+
+        def circuit():
+            qml.PauliX(wires=1)
+            qml.PauliY(wires=2)
+            qml.PauliZ(wires=0)
+
+            qml.Hadamard(wires=0)
+            qml.Hadamard(wires=1)
+            qml.Hadamard(wires=2)
+
+            qml.S(wires=0)
+            qml.T(wires=0)
+
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 0])
+
+            qml.CZ(wires=[0, 1])
+            qml.CZ(wires=[0, 2])
+
+            qml.SWAP(wires=[0, 1])
+            qml.SWAP(wires=[0, 2])
+            qml.SWAP(wires=[1, 2])
+
+            qml.CSWAP(wires=[0, 1, 2])
+
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        qjit_fn = qjit()(qml.qnode(device)(circuit))
+        qml_fn = qml.qnode(qml.device("default.qubit", wires=3))(circuit)
+
+        assert np.allclose(qjit_fn(), qml_fn())
+
+    @pytest.mark.parametrize(
+        "device",
+        [
+            qml.device(
+                "braket.local.qubit",
+                backend="braket_sv",
+                wires=2,
+            ),
+        ],
+    )
+    def test_unsupported_gate_braket(self, device):
+        """Test unsupported gates on braket devices."""
+
+        def circuit():
+            U1 = 1 / np.sqrt(2) * np.array([[1.0, 1.0], [1.0, -1.0]], dtype=complex)
+            qml.QubitUnitary(U1, wires=0)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        qjit_fn = qjit()(qml.qnode(device)(circuit))
+
+        with pytest.raises(RuntimeError):
+            # QubitUnitary is not supported!
+            qjit_fn()
```

Comment:
@maliasadi you can use

```
pytest.raises(RuntimeError, match='QubitUnitary is not supported')
```

to test the right error message is returned. If this is possible to do, it is recommended, as otherwise the test would pass on _any_ `RuntimeError`, even a different, unrelated one!